src/codewhisperer/util/authUtil.ts

```diff
@@ -142,18 +143,47 @@ export class AuthUtil {
                 throw ToolkitError.chain(err, 'Unable to authenticate connection')
             }
         }
+        await Promise.all([
+            vscode.commands.executeCommand('aws.codeWhisperer.refresh'),
+            vscode.commands.executeCommand('aws.codeWhisperer.refreshRootNode'),
+            vscode.commands.executeCommand('aws.codeWhisperer.refreshStatusBar'),
+        ])
     }
 
-    public async showReauthenticatePrompt() {
+    public async showReauthenticatePrompt(isAutoTrigger?: boolean) {
+        const doNotShowAgain =
+            globals.context.globalState.get<boolean>(CodeWhispererConstants.connectionExpiredDoNotShowAgainKey) || false
```

Comment:
Could use `PromptSettings` here:
```suggestion
        const doNotShowAgain = !(await PromptSettings.instance.isPromptEnabled('codewhispererExpiredConnection'))
```
Then use `PromptSettings.instance.disablePrompt('codewhispererExpiredConnection')` to disable it
You'll have to edit `package.json` to add the key though unfortunately. Example: https://github.com/aws/aws-toolkit-vscode/blob/40a7aa6f787b2f377ddcfde0170af99a22f8413d/package.json#L216-L219

Reply:
nice! will do that thanks

Reply:
updated